src/KeyConnector/Services/GoogleCloudKmsRsaKeyService.cs

```diff
@@ -26,14 +26,14 @@ public class GoogleCloudKmsRsaKeyService : IRsaKeyService
 
         public async Task<byte[]> EncryptAsync(byte[] data)
         {
-            var result = await _keyManagementServiceClient.EncryptAsync(_cryptoKeyName, ByteString.CopyFrom(data));
-            return result.Ciphertext.ToByteArray();
-
+            var publicKey = await GetRsaPublicKeyAsync();
+            var result = publicKey.Encrypt(data, RSAEncryptionPadding.OaepSHA256);
+            return result;
         }
 
         public async Task<byte[]> DecryptAsync(byte[] data)
         {
-            var result = await _keyManagementServiceClient.DecryptAsync(_cryptoKeyName, ByteString.CopyFrom(data));
+            var result = await _keyManagementServiceClient.AsymmetricDecryptAsync(_cryptoKeyVersionName, ByteString.CopyFrom(data));
```

Comment:
Seems like we don't need `_cryptoKeyName` anymore. Delete?

Reply:
Deleted 👍